### ISSUE TITLE
Remove deprecated t.a.app.reportProfile

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4093]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4093|
|Type|task|
|Created|2009-11-13 19:23:11Z|

```reportProfile``` in [twisted/application/app.py](twisted/application/app.py) has been deprecated since 8.0, time to get rid of it (eventually).

Attachments:

* [trac-4093.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/a79/a79141fe15a713226c2e84697e0d6c2217172524/0c6c6353456418917bff932727574f06bee83be3.patch) (581 bytes) - added by SW on 2010-05-17 12:24:16Z - removal of deprecated reportProfile()

<details><summary>Searchable metadata</summary>

```
trac-id__4093 4093
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1258140191000000 1258140191000000
changetime__1274194806000000 1274194806000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks.  This change needs a news file as well, and the copyright date on `twisted/application/app.py` should be updated as well.  These are simple, so I'll take care of them and then commit to trunk.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Oh.  But also, importantly, the test for this function needs to be deleted as well.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [28961]) Remove twisted.applications.app.reportProfile

Author: SW, exarkun
Reviewer: exarkun
Fixes: [#4093](https://github.com/twisted/twisted/issues/4093)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
